rust/agents/scraper/src/scraper/mod.rs

```diff
@@ -795,39 +824,53 @@ impl SqlChainScraper {
                 .get_mut(&hash)
                 .expect("We found a block that we did not request")
-                .insert((Some(block.id), block.timestamp));
+                .insert((
+                    Some(block.id),
+                    BlockInfo {
+                        hash,
+                        timestamp: date_time::to_unix_timestamp_s(&block.timestamp),
+                        // TODO: we don't actually use these below, we should make sure to clean
+                        // this out
+                        number: 0,
+                        gas_used: Default::default(),
+                        gas_limit: Default::default(),
+                    },
+                ));
         }
 
         let blocks_to_fetch = blocks
             .iter_mut()
             .inspect(|(_, info)| {
                 // info being defined implies the id has been set (at this point)
-                debug_assert!(info.is_none() || info.unwrap().0.is_some())
+                debug_assert!(info.is_none() || info.as_ref().unwrap().0.is_some())
             })
             .filter(|(_, block_info)| block_info.is_none());
-        for (_hash, block_info) in blocks_to_fetch {
-            // TODO: fetch block data from ethers
-            // (timestamps are not included in LogMeta)
-            let _ = block_info.insert((None, crate::date_time::now()));
+        for (hash, block_info) in blocks_to_fetch {
+            let info = self.local.provider.get_block_by_hash(hash).await?;
+            let _ = block_info.insert((None, info));
         }
 
         // insert any blocks that were not known and get their IDs
         // use this vec as temporary list of mut refs so we can update once we get back
         // the ids.
         let mut blocks_to_insert: Vec<(&H256, &mut OptionalBlockInfo)> = blocks
             .iter_mut()
-            .filter(|(_, info)| info.unwrap().0.is_none())
+            .filter(|(_, info)| info.as_ref().unwrap().0.is_none())
             .collect();
-        let models: Vec<block::ActiveModel> = blocks_to_insert
-            .iter_mut()
+
+        let mut models: Vec<block::ActiveModel> = blocks_to_insert
+            .iter()
             .map(|(hash, block_info)| {
+                let block_info = block_info.as_ref().unwrap();
                 block::ActiveModel {
                     id: NotSet,
                     hash: Set(format_h256(hash)),
-                    time_created: Set(crate::date_time::now()),
+                    time_created: Set(date_time::now()),
                     domain: Unchanged(self.local_domain() as i32),
-                    height: Unchanged(0), // TODO: get this from ethers
-                    timestamp: Set(block_info.unwrap().1),
+                    height: Unchanged(block_info.1.number as i64),
+                    timestamp: Set(date_time::from_unix_timestamp_s(block_info.1.timestamp)),
+                    gas_used: Set(u256_as_scaled_f64(block_info.1.gas_used, 18)),
```

Comment:
ditto here no need to be scaling these or to have them as floats

Reply:
See issue posted above.